File: 0.Core/Entities/Team.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Core.Entities
{
    public class Team
    {
        [Key]
        public int TeamId { get; set; }

        [Required]
        public string Name { get; set; }

        public string Country { get; set; }

        public string League { get; set; }

        public string Description { get; set; }

        public string ImageSource { get; set; }

        public Team(string Name, string Country, string League, string Description,string ImageSource)
        {
            this.Name = Name;
            this.Country = Country;
            this.League = League;
            this.Description = Description;
            this.ImageSource = ImageSource;
        }

        public Team(string Name,string Desciption,string ImageName)
        {
            this.Name = Name;
            this.Description = Description;
            this.ImageSource = ImageSource;
        }

        public Team()
        {
            this.Name = "";
            this.Country = "";
            this.League = "";
            this.Description = "";
            this.ImageSource = "";
        }

    }
}

File: 3.Presentation/ClientApp/app/models/team.ts
export interface ITeam {
    teamId: number;
    name: string;
    league: string;
    country: string;
    description: string;
    imageSource: string;
}

export class Team implements ITeam {
    constructor(
        public teamId: number,
        public name: string,
        public league: string,
        public country: string,
        public description: string,
        public imageSource: string) { }
}
